src/file.ts

```diff
@@ -287,12 +299,18 @@ interface FileQuery {
 }
 
 interface SignedUrlQuery {
-  GoogleAccessId: string;
-  Expires: number;
-  Signature: string;
-  generation: number;
-  'response-content-type': string;
-  'response-content-disposition': string;
+  GoogleAccessId?: string;
```

Comment:
Are all of these really optional?

Reply:
Good question!

v2 requires
```
  GoogleAccessId?: string;
  Expires?: number;
  Signature?: string;
```

while v4 requires
```
  'X-Goog-Algorithm'?: string;
  'X-Goog-Credential'?: string;
  'X-Goog-Date'?: string;
  'X-Goog-Expires'?: number;
  'X-Goog-SignedHeaders'?: string;
  'X-Goog-Signature'?: string;
```

Reply:
Refactored `SignedUrlQuery` to contain the shared query params.

Create interfaces `{V4|V2}SignedUrlQuery` to contain params specific to each version.